RDF.js

```diff
@@ -340,13 +351,89 @@ function detectType(newItem, node, ret) {
 				if(container = getNodeByType(isPartOf,
 						[n.bib+"CourtReporter", n.bibo+"CourtReporter"])) {
 					t.bib = "case";
+				} else if (getFirstResults(node, [n.bibo+"isbn10", n.bibo+"isbn13"], true)) {
+					t.bib = "book";
 				} else {
 					t.bib = "webpage";
 				}
 			break;
+			
+			// schema.org types
+			// subtypes of http://schema.org/CreativeWork and https://bib.schema.org/
+			case 'newsarticle':
+			case 'analysisnewsarticle':
+			case 'backgroundnewsarticle':
+			case 'opinionNewsarticle':
+			case 'reportagenewsarticle':
+			case 'reviewnewsarticle':
+				t.so = 'newspaperArticle'; break;
+			case 'scholarlyarticle':
+			case 'medicalscholarlyarticle':
+				t.so = 'journalArticle';
+				containerPublicationIssue = getNodeByType(isPartOf, [n.so+"PublicationIssue"]);
+				containerPublicationVolume = getNodeByType(isPartOf, [n.so+"PublicationVolume"]);
+				containerPeriodical = getNodeByType(isPartOf, [n.so+"Periodical"]);
+				container = getNodeByType(isPartOf, [n.so+"PublicationIssue", n.so+"PublicationVolume", n.so+"Periodical"]);
+				break;
+			case 'chapter':
+				t.so = 'bookSection';
+				container = getNodeByType(isPartOf, [n.so+"Book"]);
+				break;
+			case 'socialmediaposting':
+			case 'blogposting':
+			case 'liveblogposting':
+				t.so = 'blogPost';
+				break;
+			case 'discussionforumposting':
+				t.so = 'forumPost'; break;
+			case 'techarticle':
+			case 'apireference':
+				t.soGuess = 'report'; break;
+			case 'clip':
+			case 'movieclip':
+			case 'videogameclip':
+				t.soGuess = 'videoRecording'; break;
+			case 'tvclip':
+			case 'tvepisode':
```

Comment:
why split up this & following case?

Reply:
This case uses `t.so` and the next one `t.soGuess`. Thus, they have different order in the final calculation:
```js
	var itemType = t.zotero || t.bib || t.prism || t.eprints || t.og || t.dc ||
		t.so ||
		exports.defaultUnknownType || t.zoteroGuess || t.bibGuess ||
		t.prismGuess || t.ogGuess || t.dcGuess || t.soGuess;
```
I put "tvseries" (a whole series is maybe something else) and "episode" (this sounds generic) as a guess in the second case. Do you suggest to use `t.so` for all of them?

Reply:
Does "+1" mean it is okay like it is or should I change it?

Reply:
that it's OK